### PR TITLE
Do not retry requests that have been dropped by the user.

### PR DESCRIPTION
**Pull request by nihohit**
https://github.com/valkey-io/valkey-glide/issues/2138

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nihohit** [comment]
@barshaul @avifenesh, please ping me when you have issues that are also relevant for upstream redis-rs - I'd be happy to contribute fixes, if I'm able to, and will be even happier to upstream your fixes :)

File: redis/tests/test_cluster_async.rs [hunk 1]
```diff
@@ -4142,6 +4142,49 @@ mod cluster_async {
...
+
+                Err(Ok(Value::Okay))
+            },
+        );
+
+        runtime.block_on(async move {
+            cmd.query_async::<_, Value>(&mut connection)
+                .timeout(futures_time::time::Duration::from_millis(1))
+                .await
+                .unwrap_err();
```
Comment (barshaul) [inline redis/tests/test_cluster_async.rs:4178]: Can you add an assertion that the received error is a timeout error?

File: redis/tests/test_cluster_async.rs [hunk 2]
```diff
@@ -4142,6 +4142,49 @@ mod cluster_async {
...
+        let request_counter = Arc::new(AtomicU32::new(0));
+        let cloned_req_counter = request_counter.clone();
+        let MockEnv {
+            runtime,
+            async_connection: mut connection,
+            ..
+        } = MockEnv::with_client_builder(
+            ClusterClient::builder(vec![&*format!("redis://{name}")])
+                .retries(5)
+                .retry_wait_formula(1, 1)
```
Comment (barshaul) [inline redis/tests/test_cluster_async.rs:4159]: minor: why not using max_retry_wait with 1? I think it's more readable

**Review by barshaul — APPROVED** [review]
Minor comments.
Thanks a lot Shachar! 🌟 👑

**Comment by barshaul** [comment]
> @barshaul @avifenesh, please ping me when you have issues that are also relevant for upstream redis-rs - I'd be happy to contribute fixes, if I'm able to, and will be even happier to upstream your fixes :)

Absolutely, will do! Thanks!

**Comment by nihohit** [comment]
@barshaul done.

**Pull request merged**